osquery/tables/system/windows/groups.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#define _WIN32_DCOM
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+// clang-format off
+#include <LM.h>
+// clang-format on
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/logger.h>
+
+#include "osquery/core/process.h"
+#include "osquery/core/windows/wmi.h"
+#include "osquery/tables/system/windows/registry.h"
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+
+// Defined in /osquery/core/windows/process_ops.cpp, but not declared in a
+// header:
+std::string psidToString(PSID sid);

```

Comment:
You can add it here, https://github.com/facebook/osquery/blob/master/osquery/core/process.h

Just wrap it in an `#ifdef WIN32`.

Reply:
What about dropping it into `process_ops` alongside the other `Sid` functions?

Reply:
Following the example of `/osquery/core/windows/wmi.cpp` having its own `/osquery/core/windows/wmi.h`, I'll separate `/osquery/core/windows/process_ops.cpp` into a `cpp` and a `h` file. Then I can forward-decleare the function `psidToString()` from `process_ops.cpp` there in the `process_ops.h`.